amqp/transport.py

```diff
@@ -54,7 +55,7 @@ class _AbstractTransport(object):
     """Common superclass for TCP and SSL transports"""
     connected = False
 
-    def __init__(self, host, connect_timeout):
+    def __init__(self, host, connect_timeout, write_timeout, read_timeout):
```

Comment:
In L105 you check if `write_timeout` and `read_timeout` are None. I think it'd be worth having these 2 options as keyword arguments instead of being required.

Also, without this, we'd break backwards compatibility for consumers of py-amqp